src/test/java/seedu/address/logic/OverdueCheckerTest.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.logic;
+
+import static junit.framework.TestCase.assertEquals;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import javafx.application.Platform;
+import javafx.collections.ObservableList;
+import seedu.address.model.Model;
+import seedu.address.model.ReadOnlyDeskBoard;
+import seedu.address.model.activity.Activity;
+import seedu.address.model.activity.exceptions.ActivityNotFoundException;
+import seedu.address.model.activity.exceptions.DuplicateActivityException;
+import seedu.address.testutil.TypicalActivities;
+
+public class OverdueCheckerTest {
+
+    private ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+
+    //TODO
+    /**
+     *
+     */
+    public void run_markingAsOverdue_correctNumOverdueTasks() throws InterruptedException {
+        ModelStub modelStub = new ModelStub();
+        OverdueChecker checker = new OverdueChecker(modelStub);
+
+        executor.schedule(checker, 0, TimeUnit.SECONDS);
+        waitForRunLater();
+        assertEquals(OverdueChecker.getNumOverdueTasks(), 3);
+    }
+
+    private void waitForRunLater() throws InterruptedException {
+        Semaphore semaphore = new Semaphore(0);
+        Platform.runLater(() -> semaphore.release());
+        semaphore.acquire();
+    }
+
+    /**
+     * Model stub
+     */
+    private class ModelStub implements Model {
```

Comment:
Since model stub is being used in other tests, perhaps we can abstract this ModelStub out to be a standalone class by itself. 

This increases code quality as other tests can reuse the class without having to actually implement it again.